Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/NonprofitSignupViewController.m

```diff
@@ -68,61 +68,27 @@ - (IBAction)getStartedButtonTapped:(id)sender {
     
-    NSOperationQueue *opQueue = [NSOperationQueue new];
-
-    NSBlockOperation *signupUserOp = [NSBlockOperation new];
-    signupUserOp.queuePriority = NSOperationQueuePriorityHigh;
-    __weak NSBlockOperation *weakSignupUserOp = signupUserOp;
-    [signupUserOp addExecutionBlock:^{
-        [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-            if (!succeeded) {
-                [opQueue cancelAllOperations];
-                UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save user." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                [self presentViewController:alert animated:YES completion:nil];
-            }
-        }];
-    }];
-    
-    NSBlockOperation *saveNonprofitOp = [NSBlockOperation new];
-    saveNonprofitOp.queuePriority = NSOperationQueuePriorityHigh;
-    __weak NSBlockOperation *weakSaveNonprofitOp = saveNonprofitOp;
-    [saveNonprofitOp addExecutionBlock:^{
-        [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-            if ([weakSaveNonprofitOp isCancelled]) {
-                [self.user deleteInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-                    if (!succeeded) {
-                        NSLog(@"error");
-                    }
-                }];
-                return;
-            } else if (!succeeded) {
-                [opQueue cancelAllOperations];
-                UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save nonprofit." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                [self presentViewController:alert animated:YES completion:nil];
-            }
-        }];
-    }];
-    
-    
-    NSBlockOperation *loginSegueOp = [NSBlockOperation new];
-    //FIXME: loginSegueOp is still running first??
-    loginSegueOp.queuePriority = NSOperationQueuePriorityLow;
-    __weak NSBlockOperation *weakLoginSegueOp = loginSegueOp;
-    [loginSegueOp addExecutionBlock:^{
-        if (weakLoginSegueOp.isCancelled) {
-            return;
+    [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            [self.nonprofit saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+                if (succeeded) {
+                    [self performSegueWithIdentifier:@"loginSegue" sender:nil];
+                } else {
+                    [self.user deleteInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+                        if (!succeeded) {
+                            UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save nonprofit." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
+                            [self presentViewController:alert animated:YES completion:nil];
```

Comment:
Does the Parse SDK documentation state that these completion blocks are always called in the main thread? I  didn't found it on their API reference so I wouldn't assume it, maybe it works now but it might change in the future if it's not documented. I would manually execute this on the main thread with `[[NSOperationQueue mainQueue] addOperationWithBlock:...` or `dispatch_async(dispatch_get_main_queue(),...`.